lib/vrl/stdlib/src/parse_regex.rs

```diff
@@ -63,7 +63,7 @@ impl Expression for ParseRegexFn {
             .pattern
             .captures(&value)
             .map(|capture| util::capture_regex_to_map(&self.pattern, capture))
-            .ok_or("unable to parse regular expression")?;
+            .ok_or("could not find any pattern matches")?;
```

Comment:
Updated the error message to better reflect what happened.

Reply:
The new error message sounds a bit like "we used the regular expression, but did not find any matches". In this case, the regular expression was malformed and never run, though - or am I missing something?

Reply:
No, a regular expression cannot be malformed, that's caught by the compiler. This turns a `none` match (e.g. no capture group matched any of the input) into an error.

It was decided that that case should return an error, so that you can chain parse functions using `parse_apache_log ?? parse_regex ?? parse_...` and have the first one that matches return the value.

It's a bit of a questionable decision to return an error if the pattern doesn't match, but I can see the appeal of chaining multiple parse errors.

Because of that, I merely updated the message here to make it more clear why the error occurs.

Reply:
> The new error message sounds a bit like "we used the regular expression, but did not find any matches".

That is exactly what is happening here, so it's a good error message!